### PR TITLE
feat(IN-3538): orchestration

### PR DESCRIPTION
**Pull request by WojtekTheWebDev**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSDoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Add orchestration possibility by extending the context with `integrations` and `getIntegration` properties.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/middleware/src/handlers/prepareApiFunction/index.ts [hunk 1]
```diff
@@ -29,6 +29,40 @@ export function prepareApiFunction(
...
+          customQueries: innerCustomQueries,
+        };
+
+        const createInnerApiClient = innerApiClient.createApiClient.bind({
+          middleware: innerMiddlewareContext,
+        });
+
+        const apiClientInstance = createInnerApiClient({
+          ...innerConfiguration,
+          ...innerInitConfig,
```
Comment (michaelKurowski) [inline packages/middleware/src/handlers/prepareApiFunction/index.ts:61]: Do we preserve the methods exposed on `context.api` or do we not have to worry about them here?

Comment (WojtekTheWebDev) [reply packages/middleware/src/handlers/prepareApiFunction/index.ts:61]: Context.api method will be generated during innerCreateApiClient (same behaviour as with previous createApiClient), so nothing changes here

**Review by michaelKurowski — APPROVED** [review]
Its ok from my side but check tests firsts as you have breaking change in types.

File: packages/middleware/src/createServer.ts [hunk 2]
```diff
@@ -14,20 +14,21 @@ import {
...
   TIntegrationContext extends Record<string, IntegrationContext>
 >(config: MiddlewareConfig<TIntegrationContext>): Promise<Express> {
+  const app = express();
+
+  app.use(express.json());
+  app.use(cookieParser());
+  app.use(
+    cors({
+      credentials: true,
+      origin: "http://localhost:3000",
```
Comment (Fifciu) [inline packages/middleware/src/createServer.ts:27]: When I look at it, I feel like we should introduce a second optional argument to the createServer, where we could pass e.g. config to the cors middleware. But it's outside of the scope of this task, just a food for thought

**Review by Fifciu — APPROVED** [review]
Nitpicks to add to batch and commit, looks good. Great work Wojtek

**Review by bartoszherba — COMMENTED** [review]
@WojtekTheWebDev Are we going to build changelog based on the changeset messages? Because these messages are meaningless.

**Comment by WojtekTheWebDev** [comment]
@bartoszherba Yeah I know, I'm going to fix it as soon as I'll close the prerelease mode to not do it twice

**Pull request merged**